streamer/src/nonblocking/quic.rs

```diff
@@ -218,7 +231,24 @@ async fn run_server(
         }
 
         if let Ok(Some(connection)) = timeout_connection {
-            info!("Got a connection {:?}", connection.remote_address());
+            let remote_address = connection.remote_address();
+
+            let do_rate_limiting = true;
+            // first check overall connection rate limit:
+            if do_rate_limiting && !overall_connection_rate_limiter.check(&remote_address.ip()) {
+                stats.connection_throttled.fetch_add(1, Ordering::Relaxed);
+                continue;
+            }
+
+            info!("Got a connection {remote_address:?}");
+            if do_rate_limiting && !rate_limiter.check(&remote_address.ip()) {
+                info!(
+                    "Reject connection from {:?} -- rate limiting exceeded",
+                    remote_address
+                );
+                stats.connection_throttled.fetch_add(1, Ordering::Relaxed);
```

Comment:
Should we add a new stats to see if connections are being throttled due to per-IP limits?

Reply:
That makes sense, will do